### PR TITLE
+ parser error ~ refactoring formats

### PR DESCRIPTION
**Pull request by pape-io**

### CODE REVIEW

File: lib/lhc/request.rb [hunk 1]
```diff
@@ -17,6 +17,7 @@ def initialize(options, self_executing = true)
     generate_url_from_template!
     self.iprocessor = LHC::InterceptorProcessor.new(self)
     self.raw = create_request
+    self.format = options.delete('format') || JsonFormat.new
```
Comment (acoiro) [inline lib/lhc/request.rb]: I would call it parser instead of format, everywhere.

**Comment by acoiro** [comment]
Small comment, rest :+1:

File: lib/lhc/request.rb (hunk 1 above)
Comment (pape-io) [inline lib/lhc/request.rb]: As discussed face to face:
1. `JSON.parse` is quite common so `format.parse` is okay
2. format does more then parsing. It also wraps the request

**Comment by maveonair** [comment]
@pape-io :heart:

File: lib/lhc/formats/json.rb [hunk 2]
```diff
@@ -1,9 +1,24 @@
-module JsonFormat
+class JsonFormat
```
Comment (maveonair) [inline lib/lhc/formats/json.rb]: :+1:

File: lib/lhc/response.rb [hunk 3]
```diff
@@ -16,15 +16,7 @@ def initialize(raw, request)
...
-    @data ||=
-      # rubocop:disable Style/IdenticalConditionalBranches
-      case format
-      when :json
-        JSON.parse(raw.body, object_class: OpenStruct)
-      else # default is json
-        JSON.parse(raw.body, object_class: OpenStruct)
-      end
-    # rubocop:enable Style/IdenticalConditionalBranches
+    @data ||= format.parse(self)
```
Comment (maveonair) [inline lib/lhc/response.rb]: Very well done!

**Pull request merged**